docs/book/development.md

```diff
@@ -0,0 +1,129 @@
+go: downloading golang.org/x/tools v0.0.0-20200407143752-a3568bac92ae
+go: extracting golang.org/x/tools v0.0.0-20200407143752-a3568bac92ae
+go: downloading golang.org/x/tools v0.0.0-20190614205625-5aca471b1d59
+go: extracting golang.org/x/tools v0.0.0-20190614205625-5aca471b1d59
+go: downloading golang.org/x/mod v0.2.0
+go: extracting golang.org/x/mod v0.2.0
+go: downloading golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
+go: extracting golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
+hack/check-lint.sh
+go: finding golang.org/x/lint latest
+hack/check-mdlint.sh
+hack/check-shell.sh
+hack/check-staticcheck.sh
+...
+```
+
+## How to test CSI driver in a Kubernetes cluster
+
+- Build driver images , tag them in your local registry and then push images to your container registry.
+vSphere CSI driver includes two images:
+
+  - CSI controller (image name is "driver")
+
+  - CSI Metadata Syncer (image name is "syncer")
+
+``` sh
+make images
+go: downloading github.com/rexray/gocsi v1.0.1
+go: downloading github.com/vmware/govmomi v0.21.1-0.20191010195446-c0f4e9754e39
+go: extracting github.com/rexray/gocsi v1.0.1
+go: extracting github.com/vmware/govmomi v0.21.1-0.20191010195446-c0f4e9754e39
+go: finding github.com/rexray/gocsi v1.0.1
+go: finding github.com/vmware/govmomi v0.21.1-0.20191010195446-c0f4e9754e39
+hack/release.sh
+building gcr.io/cloud-provider-vsphere/csi/ci/driver:v0.2.1-103-ga6fc92a
+GOPROXY=https://proxy.golang.org
+Sending build context to Docker daemon  95.19MB
+Step 1/16 : ARG GOLANG_IMAGE=golang:1.12.10
+Step 2/16 : ARG BASE_IMAGE=gcr.io/cloud-provider-vsphere/extra/csi-driver-base:v0.2.1-10-gb2fb75e
+...
+
+
+Step 16/16 : ENTRYPOINT ["/bin/vsphere-csi"]
+ ---> Running in 0e85247d765b
+Removing intermediate container 0e85247d765b
+ ---> 120c16aff160
+Successfully built 120c16aff160
+Successfully tagged gcr.io/cloud-provider-vsphere/csi/ci/driver:v0.2.1-103-ga6fc92a
+building gcr.io/cloud-provider-vsphere/csi/ci/syncer:v0.2.1-103-ga6fc92a
+Sending build context to Docker daemon  95.19MB
+Step 1/14 : ARG GOLANG_IMAGE=golang:1.12.10
+Step 2/14 : ARG BASE_IMAGE=photon:2.0
+Step 3/14 : FROM ${GOLANG_IMAGE} as builder
+...
+Step 14/14 : ENTRYPOINT ["/bin/vsphere-syncer"]
+ ---> Running in 3d523ecafa6a
+Removing intermediate container 3d523ecafa6a
+ ---> 039ddcca4e18
+[Warning] One or more build-args [VERSION] were not consumed
+Successfully built 039ddcca4e18
+Successfully tagged gcr.io/cloud-provider-vsphere/csi/ci/syncer:v0.2.1-103-ga6fc92a
+```
+
+- Replace image in yaml file
+  - Replace "vmware/vsphere-csi:<vsphere_csi_ver>" in "vsphere-csi-controller-deployment.yaml" and "vsphere-csi-node-ds.yaml" with your own image urls
+
+  - Replace "vmware/syncer:<syncer_ver>" in "vsphere-csi-controller-deployment.yaml" with your own image urls
+
+  - Follow [instruction](???) to install vSphere CSI driver
+
+## Testing
+
+- Running unit test
+
+``` sh
+
+make unit
+```
+
+- Running E2E test
```

Comment:
How do we want to utilize - `docs/book/testing.md`?

CC: @chethanv28

Reply:
How about just redirect to tests/e2e README file.
Ex: https://github.com/chethanv28/vsphere-csi-driver/blob/testing-doc/docs/book/testing.md

Reply:
Fixed.